exercises/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+)
+
+type Entry struct {
+	ID          uint32
+	FirstName   string
+	LastName    string
+	PhoneNumber string
+}
+
+func newEntry() {
+	const phonebookFile = "phonebook.json"
+	var entries []Entry
+	var ID uint32
+	e := Entry{ID: ID}
+	f, err := os.Open(phonebookFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	if err := json.NewDecoder(bufio.NewReader(f)).Decode(&entries); err != nil {
+		log.Fatal(err)
+	}
+
+	scanner := bufio.NewScanner(os.Stdin)
+	fmt.Println("Enter last name:")
+	scanner.Scan()
+	e.LastName = scanner.Text()
+	fmt.Println("Enter first name:")
+	scanner.Scan()
+	e.FirstName = scanner.Text()
+	fmt.Println("Enter phone number:")
+	scanner.Scan()
+	e.PhoneNumber = scanner.Text()
+
+	for _, i := range entries {
+		if e.ID != i.ID {
```

Comment:
The IDs not necessary in a sorted order. I think you want to find max ID, and not just any. It may work in your case, but overall it isn't correct.